src/components/DrawTool/DrawTool.tsx

```diff
@@ -0,0 +1,246 @@
+/* eslint-disable no-underscore-dangle */
+import React, { useEffect, useMemo, useState } from 'react'
+import { createGlobalStyle } from 'styled-components'
+import { ascDefaultTheme, themeColor } from '@datapunt/asc-ui'
+import { v4 as uuidv4 } from 'uuid'
+import L, { LayerEvent, LeafletKeyboardEvent, Polygon, Polyline } from 'leaflet'
+import { useMapInstance } from '@datapunt/react-maps'
+import 'leaflet-draw'
+import 'leaflet-draw/dist/leaflet.draw.css'
+import DrawToolBare from './DrawToolBare'
+import { drawIcon } from '../../icons'
+import useStateRef from '../../utils/useStateRef'
+
+const GlobalStyle = createGlobalStyle`
+  .leaflet-interactive.leaflet-mouse-marker {
+    cursor: copy;
+    
+    &.leaflet-editing-icon {
+      cursor: pointer;
+    }
+  }
+`
+
+const { tooltip: tooltipPolygon } = L.drawLocal.draw.handlers.polygon
+const { tooltip: tooltipPolyline } = L.drawLocal.draw.handlers.polyline
+
+tooltipPolygon.start = 'Klik op de kaart om te beginnen'
+tooltipPolygon.cont = 'Klik op de kaart om een punt toe te voegen'
+tooltipPolygon.end =
+  'Klik op de kaart om verder te gaan of eindig met dubbelklik of klik op de eerste punt om te eindigen'
+
+tooltipPolyline.start = 'Klik op de kaart om te meten'
+tooltipPolyline.cont = 'Klik op de kaart om een lijn te maken'
+tooltipPolyline.end =
+  'Klik op de kaart om verder te gaan of klik op de laatste punt om te eindigen'
+
+type extraLayerTypes = {
+  id: string
+  editing: { _enabled: boolean; disable: () => void }
+}
+
+type PolygonType = extraLayerTypes & Polygon
+type PolylineType = extraLayerTypes & Polyline
+
+export type ExtendedLayer = PolygonType | PolylineType
+
+L.Edit.PolyVerticesEdit = L.Edit.PolyVerticesEdit.extend({
+  options: {
+    icon: drawIcon,
+  },
+})
+
+type Props = {
+  onDrawEnd?: (layer: ExtendedLayer) => void
+  onToggle?: (showDrawTool: boolean) => void
+  onDelete?: (layersInEditMode: Array<ExtendedLayer>) => void
+  openByDefault?: boolean
+}
+
+const DrawTool: React.FC<Props> = ({
+  onToggle,
+  onDelete,
+  onDrawEnd,
+  openByDefault,
+}) => {
+  const [inEditMode, setInEditMode] = useState(false)
+  const [inCreateMode, setInCreateMode, inCreateModeRef] = useStateRef(false)
+  const [showDrawTool, setShowDrawTool] = useStateRef(openByDefault || false)
+
+  const mapInstance = useMapInstance() as L.DrawMap
```

Comment:
We could also change the `useMapInstance` function to accept a generic type that extends the Map type.

```ts
function useMapInstance<T extends L.Map = L.Map>(): T {
  // ...
}

const mapInstance = useMapInstance<L.DrawMap>()
```

Reply:
yes, we should do this in react-maps!